Hw4

unlessiamwrong opened the pull request.

unlessiamwrong committed:
Initial commit

venommile commented on migrations/v1-create-user_links.sql:
```diff
@@ -0,0 +1,10 @@
+(
+    user_id    bigint,
+    link_id    bigint,
+    created_at timestamp with time zone not null,
```
А тут точно  нужны эти поля, если они есть в links?

venommile commented on scrapper/src/test/resources/master.xml:
```diff
@@ -0,0 +1,13 @@
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd
+       http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <include file="v1-create-users.sql"/>
```
блин, xml. мне в liquibase ваще не нравится(
мб переедешь на yaml пока не поздно? -- но это по желанию).

Скачай плагин JpaBuddy/почитай про него гайды.Это сильно облегчит дальнейшую жизнь

venommile commented on scrapper/src/test/resources/v1-create-links.sql:
```diff
@@ -0,0 +1,10 @@
+(
+    id         bigint generated always as identity,
+    name       text                     not null,
+    value      text                     not null,
```
точно не какой-нибудь varchar(200). Мы не хотим сделать ограничение на длину, чтобы нас злоумышленники случайно не убили?

venommile commented on compose.yml:
```diff
@@ -2,7 +2,7 @@ services:
   postgresql:
```
Почитай про spring boot docker compose https://youtu.be/23oLE1ToURk?si=jZbxbPOscSRAn7m8, в будущем сильно поможет

unlessiamwrong committed:
Fix:
1. Change master.xml to master.yml
2. Remove unnecessary columns in tables
3. Change column type from text to varchar(200) in all tables

unlessiamwrong committed:
Fix: Change INSERT params in LiquibaseTest

unlessiamwrong committed:
Merge branch 'main' into Hw4

unlessiamwrong committed:
Merge branch 'main' into Hw4
